client/src/components/SignupForm.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import {
+  Form, Input, Button, message,
+} from 'antd';
+import { Link } from 'react-router-dom';
+import httpService from '../services/httpService';
+import Img from './common/Img';
+import ParagraphText from './common/Paragraph';
+import HeaderTitle from './common/Title';
+import 'antd/dist/antd.css';
+import './signup.css';
+
+function SignupForm() {
+  const [errorMsg, setErrorMsg] = useState('');
+  const errorlert = (erroMsg) => {
+    message.error(erroMsg);
+  };
+  const postData = async (userInfo) => {
+    try {
+      const { msg } = await httpService.post('/api/v1/signup', userInfo);
+      setErrorMsg(msg);
+    } catch (error) {
+      setErrorMsg(error);
```

Comment:
here check if the error is expected or not 
if (error.status > 400 && error.status < 500} {
set ErrorMsg...
}

you must use useEffect and stop request in return function.
take a look at service file, to stop request

Reply:
done